RHEL6_7/system/pam/check

```diff
@@ -4,22 +4,57 @@
 
 #END GENERATED SECTION
 
-POSTUPGRADE_DIR="$VALUE_TMP_PREUPGRADE/postupgrade.d/pam"
+POSTUPGRADE_DIR="$POSTUPGRADE_DIR/pam"
+CLEANCONF_DIR="$VALUE_TMP_PREUPGRADE/cleanconf/etc/pam.d"
+
 if [[ ! -d "$POSTUPGRADE_DIR" ]]; then
     mkdir -p "$POSTUPGRADE_DIR"
 fi
+if [[ ! -d "$CLEANCONF_DIR" ]]; then
+    mkdir -p "$CLEANCONF_DIR"
+fi
+
 SCRIPT_NAME="postupgrade-pam.sh"
 POST_SCRIPT="postupgrade.d/$SCRIPT_NAME"
+
 cp -f $POST_SCRIPT $POSTUPGRADE_DIR/$SCRIPT_NAME
+
 fail=0
 
+comment_invalid () {
+    perl -p -i -e 's/^/#/g if /^(?!.*#)(?=.*(pam_passwdqc|pam_ecryptfs))/;' "$1"
+}
+get_deprec_dirctv () {
+
+    deprec_dirctv="$1"
+    file="$2"
+    export deprec_dirctv
+    export file 
+    deprec_status=$(perl -ne '$deprec_dirctv = $ENV{deprec_dirctv};$file = $ENV{file}; 
+    print "PAM: The $file file contains the $deprec_dirctv module, which is no longer supported.\n"
+    if /^(?!.*#)(?=.*$deprec_dirctv)/' "$2")
+
+}
+copy_clean () {
+    cp -p "$1" "$2"
+}
+
 #pam_passwdqc and pam_ecryptfs were removed
-for file in /etc/pam.d/*;
+for file in $(find -P /etc/pam.d/ -maxdepth 1 -type f)
 do
-  grep pam_passwdqc "$file" | grep -q -v ^# && sed -e '/pam_passwdqc/s/^/#/g' >"$VALUE_TMP_PREUPGRADE/cleanconf/$file" && log_medium_risk "PAM: The $file file contains the pam_passwdqc module, which is no longer supported." && fail=1
-  grep pam_ecryptfs "$file" | grep -q -v ^# && sed -e '/pam_ecryptfs/s/^/#/g' >"$VALUE_TMP_PREUPGRADE/cleanconf/$file" && log_high_risk "PAM: The $file file contains the pam_ecryptfs module, which is no longer supported. Your ecryptfs encrypted files will not work after the upgrade. Decrypt the files before the upgrade and possibly switch to a different encryption." && fail=1
+    clean_file="$CLEANCONF_DIR"/$(basename "$file")
+
+    get_deprec_dirctv "pam_passwdqc" "$file"
```

Comment:
Here you would set the `deprec_dirctv` variable:
```
  deprec_status=$(get_deprec_dirctv "pam_passwdqc" "$file")
```

Reply:
Fixed the variable naming, so this is no longer an issue

Reply:
This is still issue as this is not related to variable naming.